### PR TITLE
FailoverClusters Module Test-Cluster Update

### PR DESCRIPTION
**Pull request by robinharwood**
Updating the Test-Cluster articles to include an alert note.

### CODE REVIEW

**Comment by robinharwood** [comment]
#sign-off

Hey @sdwheeler are you able to peer review / sign off this PR? I've added a note box to capture a key behavior, however most of the changes are to bring the article in-line with the PowerShell quality controls.

Thanks,

Robin

File: docset/winserver2022-ps/failoverclusters/Test-Cluster.md [hunk 1]
```diff
@@ -16,168 +16,90 @@ Runs validation tests for failover cluster hardware and settings.
 ## SYNTAX
 
 ```
-Test-Cluster [[-Node] <StringCollection>] [-Disk <Object[]>] [-Pool <Object[]>] [-ReportName <String>] [-List]
- [-Include <StringCollection>] [-Ignore <StringCollection>] [-Force] [-InputObject <PSObject>]
- [-Cluster <String>] [-WhatIf] [-Confirm] [<CommonParameters>]
+Test-Cluster [[-Node] <StringCollection>] [-Disk <Object[]>] [-Pool <Object[]>]
+[-ReportName <String>] [-List] [-Include <StringCollection>] [-Ignore <StringCollection>] [-Force]
+[-InputObject <PSObject>] [-Cluster <String>] [-WhatIf] [-Confirm] [<CommonParameters>]
```
Comment (sdwheeler) [inline docset/winserver2022-ps/failoverclusters/Test-Cluster.md:21]: You need a leading space on each addtitional line in the syntax block for it to render properly.
```suggestion
Test-Cluster [[-Node] <StringCollection>] [-Disk <Object[]>] [-Pool <Object[]>]
 [-ReportName <String>] [-List] [-Include <StringCollection>] [-Ignore <StringCollection>] [-Force]
 [-InputObject <PSObject>] [-Cluster <String>] [-WhatIf] [-Confirm] [<CommonParameters>]
```

**Review by sdwheeler — DISMISSED** [review]
See one suggestion. Everything else looks good. One nit to pick. Add a blank line between blocks (eg. after headers with the exception of SYNOPSIS).

**Comment by robinharwood** [comment]
Thanks @sdwheeler, update made and ready for #sign-off

File: docset/winserver2012-ps/failoverclusters/Test-Cluster.md [hunk 2]
```diff
@@ -310,17 +252,21 @@ Accept wildcard characters: False
...
-Otherwise, the specified storage pool must be taken offline before the storage tests.
-If the **Pool** parameter is not specified, storage tests run on all storage pools that are available for use in the cluster or that are in the cluster resource offline or failed state.
-The acceptable values for this parameter are:
 
- -- System.String: A string that represents the name of the clustered storage pool or pools. 
+Specifies the clustered storage pool or pools for which to run the cmdlet. When the specified
+storage pool is online and a virtual disk in the storage pool is assigned to a clustered role or
+Cluster Shared Volume, you must also specify the **Force** parameter to take the storage pool
+offline for the duration of the storage tests. Otherwise, the specified storage pool must be taken
+offline before the storage tests. If the **Pool** parameter is not specified, storage tests run on
```
Comment (michaeltlombardi) [inline docset/winserver2012-ps/failoverclusters/Test-Cluster.md:260]: Clarification question: If I don't specify **Force** and the disks are online, does the cmdlet error?

File: docset/winserver2012-ps/failoverclusters/Test-Cluster.md [hunk 3]
```diff
@@ -216,6 +152,7 @@ Accept wildcard characters: False
 ```
 
 ### -Force
+
 Runs the cmdlet without prompting for confirmation.
 By default the cmdlet will ask for confirmation from the user before proceeding.
```
Comment (michaeltlombardi) [inline docset/winserver2012-ps/failoverclusters/Test-Cluster.md:157]: Elsewhere in the documentation it calls out that the **Force** parameter takes the disks offline for the duration of the tests if needed. I think that's worth pointing out here, too.

**Review by michaeltlombardi — COMMENTED** [review]
Clarification question and suggestion, but otherwise this looks _great_!

File: docset/winserver2012-ps/failoverclusters/Test-Cluster.md (hunk 3 above)
Comment (robinharwood) [reply docset/winserver2012-ps/failoverclusters/Test-Cluster.md:157]: Good spot, I've updated this in all versions.

File: docset/winserver2012-ps/failoverclusters/Test-Cluster.md (hunk 2 above)
Comment (robinharwood) [reply docset/winserver2012-ps/failoverclusters/Test-Cluster.md:260]: Good questions, I'm just waiting on the dev team to confirm

File: docset/winserver2012-ps/failoverclusters/Test-Cluster.md [hunk 4]
```diff
@@ -187,21 +120,24 @@ Accept wildcard characters: False
...
+The acceptable values for this parameter are:
 
- -- CimInstance#MSFT_Disk: An object returned from Get-Disk, from the Windows PowerShell® storage module.
+- **Int32, Int64, Uint32, Uint64**. A number that represents a master boot record (MBR) signature of
+  the disk.
+- **System.String**. A string that represents a master boot record (MBR) signature of the disk,
+  hexadecimal format is supported.
+- **System.String**. A string that represents the GUID of a GPT disk.
+- **ClusterResource**. A cluster resource object that represents a clustered disk.
+- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell® storage
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/failoverclusters/Test-Cluster.md:139]:
```suggestion
- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell storage
```

File: docset/winserver2012-ps/failoverclusters/Test-Cluster.md [hunk 5]
```diff
@@ -232,8 +170,9 @@ Accept wildcard characters: False
 ```
 
 ### -Ignore
-Specifies which tests or category of tests to ignore during the validation test run.
-All others tests or category of tests will run.
+
+Specifies which tests or category of tests to ignore during the validation test run. All others
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/failoverclusters/Test-Cluster.md:174]:
```suggestion
Specifies which tests or category of tests to ignore during the validation test run. All other
```

File: docset/winserver2012-ps/failoverclusters/Test-Cluster.md [hunk 6]
```diff
@@ -365,7 +314,8 @@ Accept wildcard characters: False
 ```
 
 ### -WhatIf
-Shows what would happen if the cmdlet runs.
+
+Shows what would happen if the cmdlet runs. The cmdlet is not run.
 The cmdlet is not run.
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/failoverclusters/Test-Cluster.md:319]:
```suggestion
```

File: docset/winserver2012r2-ps/failoverclusters/Test-Cluster.md [hunk 7]
```diff
@@ -204,21 +138,24 @@ Accept wildcard characters: False
...
+The acceptable values for this parameter are:
 
- -- CimInstance#MSFT_Disk: An object returned from Get-Disk, from the Windows PowerShell® storage module.
+- **Int32, Int64, Uint32, Uint64**. A number that represents a master boot record (MBR) signature of
+  the disk.
+- **System.String**. A string that represents a master boot record (MBR) signature of the disk,
+  hexadecimal format is supported.
+- **System.String**. A string that represents the GUID of a GPT disk.
+- **ClusterResource**. A cluster resource object that represents a clustered disk.
+- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell® storage
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/failoverclusters/Test-Cluster.md:157]:
```suggestion
- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell storage
```

File: docset/winserver2012r2-ps/failoverclusters/Test-Cluster.md [hunk 8]
```diff
@@ -233,8 +170,10 @@ Accept wildcard characters: False
 ```
 
 ### -Force
-Runs the cmdlet without prompting for confirmation.
-By default the cmdlet will ask for confirmation from the user before proceeding.
+
+Runs the cmdlet without prompting for confirmation. By default the cmdlet will ask for confirmation
+from the user before proceeding. When combine with the **Disk** or **Pool** parameters, the disks or
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/failoverclusters/Test-Cluster.md:175]:
```suggestion
from the user before proceeding. When combined with the **Disk** or **Pool** parameters, the disks or
```

File: docset/winserver2012r2-ps/failoverclusters/Test-Cluster.md [hunk 9]
```diff
@@ -249,8 +188,9 @@ Accept wildcard characters: False
 ```
 
 ### -Ignore
-Specifies which tests or category of tests to ignore during the validation test run.
-All others tests or category of tests will run.
+
+Specifies which tests or category of tests to ignore during the validation test run. All others
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/failoverclusters/Test-Cluster.md:192]:
```suggestion
Specifies which tests or category of tests to ignore during the validation test run. All other
```

File: docset/winserver2016-ps/failoverclusters/Test-Cluster.md [hunk 10]
```diff
@@ -207,22 +139,24 @@ Accept wildcard characters: False
...
-A cluster resource object that represents a clustered disk. 
-- **CimInstance#MSFT_Disk**.
-An object returned from Get-Disk, from the Windows PowerShell® storage module.
+- **Int32, Int64, Uint32, Uint64**. A number that represents a master boot record (MBR) signature of
+  the disk.
+- **System.String**. A string that represents a master boot record (MBR) signature of the disk,
+  hexadecimal format is supported.
+- **System.String**. A string that represents the GUID of a GPT disk.
+- **ClusterResource**. A cluster resource object that represents a clustered disk.
+- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell® storage
```
Comment (JohanFreelancer9) [inline docset/winserver2016-ps/failoverclusters/Test-Cluster.md:158]:
```suggestion
- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell storage
```

File: docset/winserver2016-ps/failoverclusters/Test-Cluster.md [hunk 11]
```diff
@@ -237,7 +171,10 @@ Accept wildcard characters: False
 ```
 
 ### -Force
-Forces the command to run without asking for user confirmation.
+
+Runs the cmdlet without prompting for confirmation. By default the cmdlet will ask for confirmation
+from the user before proceeding. When combine with the **Disk** or **Pool** parameters, the disks or
```
Comment (JohanFreelancer9) [inline docset/winserver2016-ps/failoverclusters/Test-Cluster.md:176]:
```suggestion
from the user before proceeding. When combined with the **Disk** or **Pool** parameters, the disks or
```

File: docset/winserver2016-ps/failoverclusters/Test-Cluster.md [hunk 12]
```diff
@@ -252,8 +189,9 @@ Accept wildcard characters: False
 ```
 
 ### -Ignore
-Specifies which tests or category of tests to ignore during the validation test run.
-All others tests or category of tests will run.
+
+Specifies which tests or category of tests to ignore during the validation test run. All others
```
Comment (JohanFreelancer9) [inline docset/winserver2016-ps/failoverclusters/Test-Cluster.md:193]:
```suggestion
Specifies which tests or category of tests to ignore during the validation test run. All other
```

File: docset/winserver2019-ps/failoverclusters/Test-Cluster.md [hunk 13]
```diff
@@ -207,22 +139,24 @@ Accept wildcard characters: False
...
-A cluster resource object that represents a clustered disk. 
-- **CimInstance#MSFT_Disk**.
-An object returned from Get-Disk, from the Windows PowerShell® storage module.
+- **Int32, Int64, Uint32, Uint64**. A number that represents a master boot record (MBR) signature of
+  the disk.
+- **System.String**. A string that represents a master boot record (MBR) signature of the disk,
+  hexadecimal format is supported.
+- **System.String**. A string that represents the GUID of a GPT disk.
+- **ClusterResource**. A cluster resource object that represents a clustered disk.
+- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell® storage
```
Comment (JohanFreelancer9) [inline docset/winserver2019-ps/failoverclusters/Test-Cluster.md:158]:
```suggestion
- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell storage
```

File: docset/winserver2019-ps/failoverclusters/Test-Cluster.md [hunk 14]
```diff
@@ -237,7 +171,10 @@ Accept wildcard characters: False
 ```
 
 ### -Force
-Forces the command to run without asking for user confirmation.
+
+Runs the cmdlet without prompting for confirmation. By default the cmdlet will ask for confirmation
+from the user before proceeding. When combine with the **Disk** or **Pool** parameters, the disks or
```
Comment (JohanFreelancer9) [inline docset/winserver2019-ps/failoverclusters/Test-Cluster.md:176]:
```suggestion
from the user before proceeding. When combined with the **Disk** or **Pool** parameters, the disks or
```

File: docset/winserver2019-ps/failoverclusters/Test-Cluster.md [hunk 15]
```diff
@@ -252,8 +189,9 @@ Accept wildcard characters: False
 ```
 
 ### -Ignore
-Specifies which tests or category of tests to ignore during the validation test run.
-All others tests or category of tests will run.
+
+Specifies which tests or category of tests to ignore during the validation test run. All others
```
Comment (JohanFreelancer9) [inline docset/winserver2019-ps/failoverclusters/Test-Cluster.md:193]:
```suggestion
Specifies which tests or category of tests to ignore during the validation test run. All other
```

File: docset/winserver2022-ps/failoverclusters/Test-Cluster.md [hunk 16]
```diff
@@ -207,22 +139,24 @@ Accept wildcard characters: False
...
-A cluster resource object that represents a clustered disk. 
-- **CimInstance#MSFT_Disk**.
-An object returned from Get-Disk, from the Windows PowerShell® storage module.
+- **Int32, Int64, Uint32, Uint64**. A number that represents a master boot record (MBR) signature of
+  the disk.
+- **System.String**. A string that represents a master boot record (MBR) signature of the disk,
+  hexadecimal format is supported.
+- **System.String**. A string that represents the GUID of a GPT disk.
+- **ClusterResource**. A cluster resource object that represents a clustered disk.
+- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell® storage
```
Comment (JohanFreelancer9) [inline docset/winserver2022-ps/failoverclusters/Test-Cluster.md:158]:
```suggestion
- **CimInstance#MSFT_Disk**. An object returned from Get-Disk, from the Windows PowerShell storage
```

File: docset/winserver2022-ps/failoverclusters/Test-Cluster.md [hunk 17]
```diff
@@ -237,7 +171,10 @@ Accept wildcard characters: False
 ```
 
 ### -Force
-Forces the command to run without asking for user confirmation.
+
+Runs the cmdlet without prompting for confirmation. By default the cmdlet will ask for confirmation
+from the user before proceeding. When combine with the **Disk** or **Pool** parameters, the disks or
```
Comment (JohanFreelancer9) [inline docset/winserver2022-ps/failoverclusters/Test-Cluster.md:176]:
```suggestion
from the user before proceeding. When combined with the **Disk** or **Pool** parameters, the disks or
```

File: docset/winserver2022-ps/failoverclusters/Test-Cluster.md [hunk 18]
```diff
@@ -252,8 +189,9 @@ Accept wildcard characters: False
 ```
 
 ### -Ignore
-Specifies which tests or category of tests to ignore during the validation test run.
-All others tests or category of tests will run.
+
+Specifies which tests or category of tests to ignore during the validation test run. All others
```
Comment (JohanFreelancer9) [inline docset/winserver2022-ps/failoverclusters/Test-Cluster.md:193]:
```suggestion
Specifies which tests or category of tests to ignore during the validation test run. All other
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review - @robinharwood, a few minor suggested changes, please.

**Review by JohanFreelancer9 — APPROVED** [review]

File: docset/winserver2012-ps/failoverclusters/Test-Cluster.md (hunk 2 above)
Comment (robinharwood) [reply docset/winserver2012-ps/failoverclusters/Test-Cluster.md]: Wording updating to reflect the tested behaviour.

**Comment by robinharwood** [comment]
Thanks @michaeltlombardi and @JohanFreelancer9 - Updates applied, I believe this is now ready to merge. #sign-off

**Review by michaeltlombardi — APPROVED** [review]

**Pull request merged**